homeassistant/components/google_air_quality/config_flow.py

```diff
@@ -0,0 +1,139 @@
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle Google Air Quality OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+
+    _oauth_data: dict[str, Any]
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    @property
+    def extra_authorize_data(self) -> dict[str, Any]:
+        """Extra data that needs to be appended to the authorize url."""
+        return {
+            "scope": OAUTH2_SCOPE,
+            # Add params to ensure we get back a refresh token
+            "access_type": "offline",
+            "prompt": "consent",
+        }
+
+    async def async_oauth_create_entry(self, data: dict[str, Any]) -> ConfigFlowResult:
+        """Store OAuth token data and prompt user to confirm or change coordinates."""
+        self._oauth_data = data
+        token_info = data[CONF_TOKEN]
+        scopes = token_info.get("scope", "")
+        scope_list = scopes.split()
+
+        if OAUTH2_SCOPE not in scope_list:
+            return self.async_abort(
+                reason="access_not_configured",
+                description_placeholders={
+                    "message": f"Missing required scope: {OAUTH2_SCOPE}"
+                },
+            )
+
+        return self._show_form_user()
+
+    async def async_step_coordinates(
+        self, user_input: dict[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle coordinate input and create the config entry."""
+        session = aiohttp_client.async_get_clientsession(self.hass)
+        auth = api.AsyncConfigFlowAuth(
+            session, self._oauth_data[CONF_TOKEN][CONF_ACCESS_TOKEN]
+        )
+        client = GoogleAirQualityApi(auth)
+        location = user_input[CONF_LOCATION]
+        lat = location[CONF_LATITUDE]
+        lon = location[CONF_LONGITUDE]
+
+        try:
+            user_resource_info = await client.async_air_quality(lat, lon)
+        except NoDataForLocationError:
+            return self._show_form_user(
+                user_input,
+                errors={"base": "no_data_for_location"},
+            )
+        except GoogleAirQualityApiError as ex:
+            return self.async_abort(
+                reason="access_not_configured",
+                description_placeholders={"message": str(ex)},
+            )
+        except Exception:
+            self.logger.exception("Unknown error occurred")
+            return self.async_abort(reason="unknown")
+
+        unique_id = f"{lat}_{lon}"
+
+        await self.async_set_unique_id(unique_id)
+        self._oauth_data.update(
+            {
+                CONF_LATITUDE: lat,
+                CONF_LONGITUDE: lon,
+                "region_code": user_resource_info.region_code,
+            }
+        )
+        self._abort_if_unique_id_configured()
+        return self.async_create_entry(
+            title=f"Coordinates {lat}, {lon}", data=self._oauth_data
+        )
```

Comment:
Would it make sense to use subentries here? This way we can use the same account for multiple locations

Reply:
Yes, I have implemented it [here](https://github.com/home-assistant/core/pull/145237/commits/ffb30343c287fd07819bf81139de0a44cffd0e21).
I haven't used subentries before and didn't find any examples in combination with a coordinator. So I'm happy about feedback.